ONSdigital/eq-author-app
# Ear 364 unique qcodes

### What is the context of this PR?

This PR refactors the Qcode table to make it a bit easier to follow. It also provides validation message around the qCode input if there are duplicate qCodes present within the table.

### How to review 

- Create questionnaire
- Add all the answer types to one question page
- Then create individual questions for each answer type
- Go to qCode page and add a qcode to each input
- Refresh the page to check they all save (i've been using 1-x to check for gaps)
- Start adding duplicates and check the validation message appears
- Validation shouldn't fire when the answer is `""` or `null`.

Not quite sure why the code coverage is less, not sure where I need to add coverage. I don't seem to be able to add coverage to the memo'd Row component.

Also I've tried rebasing this PR and it's pretty awkward. If it's okay, I would prefer not to please.
